src/libsystemd/sd-journal/journal-file.c

```diff
@@ -2026,6 +2026,12 @@ static int journal_file_append_entry_internal(
         assert(items || n_items == 0);
         assert(ts);
 
+        if (ts->realtime < le64toh(f->header->tail_entry_realtime))
+                return log_debug_errno(SYNTHETIC_ERRNO(ETXTBSY),
+                                       "Realtime timestamp %" PRIu64 " smaller than previous realtime "
+                                       "timestamp %" PRIu64 ", refusing entry.",
+                                       ts->realtime, le64toh(f->header->tail_entry_realtime));
```

Comment:
Hmm,,, it sounds that this introduces too many log rotations when an NTP client is running. No?

Reply:
The normal write path is already covered by this: https://github.com/systemd/systemd/blob/f4a4493eb57bffec091c8f0d4bb846850e544aa4/src/journal/journald-server.c#L837-L845

This change is mostly intended as a defense-in-depth measure and extend it to cover runtime-to-persistent log flushing at boot time.

Reply:
This is unrelated to the CI failure caused by `systemd-journal-remote`, which is fixed in #24996. Behavior for `systemd-journald` should remain unchanged.